src/main/java/io/jenkins/plugins/opentelemetry/servlet/StaplerInstrumentationServletFilter.java

```diff
@@ -87,174 +97,165 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
     }
 
     public void _doFilter(HttpServletRequest servletRequest, HttpServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
-        String pathInfo = servletRequest.getPathInfo();
-        List<String> pathInfoTokens = Collections.list(new StringTokenizer(pathInfo, "/")).stream()
-            .map(token -> (String) token)
-            .filter(t -> !t.isEmpty())
-            .collect(Collectors.toList());
-
-        if (pathInfoTokens.isEmpty()) {
-            pathInfoTokens = Collections.singletonList("");
-        }
 
-        String rootPath = pathInfoTokens.get(0);
-        // The matched route (path template).
-        String httpRoute;
-        if (SKIP_PATHS.contains(rootPath)) {
-            // skip
-            filterChain.doFilter(servletRequest, servletResponse);
-            return;
-        }
-        if (rootPath.equals("$stapler")) {
-            // TODO handle URL pattern /$stapler/bound/ec328aeb-26be-43da-94a3-59f2d683131c/news
-            filterChain.doFilter(servletRequest, servletResponse);
-            return;
-        }
-        final SpanBuilder spanBuilder;
+        // Attributes common to Http Server span and metric
+        AttributesBuilder httpServerMetricOnStartAttributesBuilder = Attributes.builder()
+            .put(HttpAttributes.HTTP_REQUEST_METHOD, servletRequest.getMethod())
+            .put(UrlAttributes.URL_SCHEME, servletRequest.getScheme())
+            .put(ServerAttributes.SERVER_ADDRESS, servletRequest.getServerName())
+            .put(ServerAttributes.SERVER_PORT, (long) servletRequest.getServerPort());
+
+        String sanitizedUrl = servletRequest.getScheme() + "://" + servletRequest.getServerName() + ":" + servletRequest.getServerPort() + servletRequest.getRequestURI() + (servletRequest.getQueryString() != null ? "?" + servletRequest.getQueryString() : "");
```

Comment:
Should we sanitize the URL first? 

cc @Wadeck

Reply:
Great catch. I completely removed the query string with a comment in the code explaining the rationale: we diverge from the OTel specs omitting the query  string rather than redacting it because it's too complicated for us to maintain proper redaction.
Instead, we will capture explicitly interesting query parameters and users can use the config `otel.instrumentation.servlet.experimental.capture-request-parameters` if we forgot something.

Reply:
Thanks for the change. Due to the code being changed, I will not dig into the topic 👍 if we feel you need a security review, feel free to ask ;)